Handle simple NNAPI flatten NHWC case

axitkhurana opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #61796

Summary:

We can easily handle nnapi conversion for nhwc inputs
that have 1 channel or H & W are 1

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29827735](https://our.internmc.facebook.com/intern/diff/D29827735)

Commit:
Handle simple NNAPI flatten NHWC case

Summary:

We can easily handle nnapi conversion for
the times we flatten all dimensions

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "Handle simple NNAPI flatten NHWC case"


Summary:

We can easily handle nnapi conversion for nhwc inputs
that have 1 channel or H & W are 1

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29827735).

dreiss commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -940,9 +940,12 @@ def add_flatten(self, node):
+        if in_oper.dim_order != DimOrder.PRESUMED_CONTIGUOUS and not is_trivial_flatten:
             raise Exception(
-                "Currently, reshape is not supported on NHWC tensors")
+                "Currently, flatten is not supported on NHWC tensors")
```
I would update this message, adding something like "unless C=1 or H=W=1"

Commit:
Update on "Handle simple NNAPI flatten NHWC case"


Summary:

We can easily handle nnapi conversion for nhwc inputs
that have 1 channel or H & W are 1

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29827735](https://our.internmc.facebook.com/intern/diff/D29827735)

[ghstack-poisoned]

axitkhurana commented:
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29827735).

Commit:
Update on "Handle simple NNAPI flatten NHWC case"


Summary:

We can easily handle nnapi conversion for nhwc inputs
that have 1 channel or H & W are 1

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_flatten

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29827735](https://our.internmc.facebook.com/intern/diff/D29827735)

[ghstack-poisoned]
